### PR TITLE
feat: add intro blog on LLM agents

### PR DESCRIPTION
**Pull request by Cali0707**
This adds an introductory blog about LLM agents with Knative (based on work that can be found in https://github.com/keventmesh/llm-tool-provider)

### CODE REVIEW

File: blog/docs/articles/llm-agents-overview.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+that information to make decisions and take actions to accomplish a goal. In the context of LLMs, an Agent
+is a system with an LLM at its core that is able to make decisions on what actions to take as it works to
+answer the prompt it received. The most common actions LLM agents can be built to take are: sending text
+or other media to the user, calling a tool to help answer the user, and calling another agent to help answer
+the user. Generally speaking, an LLM agent will also have a system prompt explaining what it’s role is and
+giving it some rules over when to call tools and/or reply to the user. For most Agents, the control flow
+can be shown as follows:
+
+<figure markdown="1">
+![image](./images/llm-agent-flow.webp)
```
Comment (matzew) [inline blog/docs/articles/llm-agents-overview.md:29]: does not render in preview site

File: blog/docs/articles/llm-agents-overview.md [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+## How do LLMs call Tools?
+
+All that is needed for an LLM to call a tool is some way for the LLM to communicate which tool it wants to call,
+as well as what arguments (if any) it wants to provide to the tool. As the LLM output is just a sequence of tokens
+there needs to be some kind of external system that can parse this information from the output, so the LLM needs
+to output structured or semi structured data in a consistent way to call a tool. There are multiple different APIs
+on how to do this, and it currently somewhat depends on the exact model you are using (as they have been trained
+to handle this in different ways). For the purposes of this blog, we are going to be specifically looking at the
+OpenAI Chat API for doing this, however the concepts are generally the same for other models.
```
Comment (matzew) [inline blog/docs/articles/llm-agents-overview.md:48]: link to the api?

File: blog/docs/articles/llm-agents-overview.md (hunk 1 above)
Comment (Cali0707) [reply blog/docs/articles/llm-agents-overview.md:29]: ah, we were trying to center it, guess this doesn't work :(

File: blog/docs/articles/llm-agents-overview.md [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+
+1. Create a trigger from your broker to your service using a filter selecting a specific type of event
+2. Create an EventType with the same `type` that uses the `schemaData` to describe the API of your service
+
+## Wrapping up
+
+In this blog post, we have covered what LLM agents are, how LLMs can call tools, and how Knative can be used to simplify
+tool calling using metadata. If you are interested in how to build an LLM agent system that uses Knative to simplify
+tool calling and discovery, as well as our vision for how this technology should evolve going forwards, keep your eyes
+out for our next blog on this topic which is coming soon! If you have any questions on this topic, we invite you to engage
```
Comment (varshith257) [inline blog/docs/articles/llm-agents-overview.md:168]: Maybe it could be good if we add slack link here :)

**Comment by varshith257** [comment]
Excited to see LLM agents blog on Knative soon

**Pull request merged**